uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/services/document.js

```diff
@@ -242,6 +272,41 @@ define([
           });
 
           return deferred.promise;
+        },
+        CacheContactsAndAssignments: function (documentList, options) {
```

Comment:
whenever you write a new method, always think first how people would use it. Even better, how *you* would like the method to work if you were to just be the "consumer" of the method, and not the author. Right now it just looks like this
```js
DocumentService.CacheContactsAndAssignments(myDocuments, ['contacts', 'assignments']);
```
If you put yourself in the shoes of a developer that would use your method, several questions would pop up

1. Why is the method beginning with a capital letter? This looks like a constructor method!
2. The method is called "cache contacts and assignments", why do I need to specify, in the second param, that i want to cache contacts and assignments? It's already there in the name, i would expect the method to do it automatically! What would be natural to expect is that by default it caches both contacts and assignments
    ```js
    DocumentService.CacheContactsAndAssignments(myDocuments);
   ```
    but that, if i wanted, i could say explicitly that i want to cache only one of the two
    ```js
    DocumentService.CacheContactsAndAssignments(myDocuments, 'contacts');
   ```
3. How do i know when the method is done? It's performing an async operation, but it's not returning a promise or anything! I would expect to be able to do something like this
    ```js
    DocumentService.CacheContactsAndAssignments(myDocuments).then(function () {
      // they are cached, moving on...
    });
   ```